fdbrpc/sim2.actor.cpp

```diff
@@ -1847,6 +1848,11 @@ Future< Void > Sim2FileSystem::deleteFile( std::string filename, bool mustBeDura
 	return Sim2::deleteFileImpl(&g_sim2, filename, mustBeDurable);
 }
 
+Future<Void> Sim2FileSystem::renameFile(std::string const& from, std::string const& to) {
+	::renameFile(from, to);
```

Comment:
We should also have a deterministic delay before actually doing the rename in simulation

Reply:
why? The rename itself is blocking...

Reply:
We might also consider fault injection here, but that's probably beyond the scope of this PR

Reply:
This also shares the same deficiency as https://github.com/apple/foundationdb/blob/d2bd63d0cbf2f0c39b748f1f39395dbda4c31a18/fdbrpc/sim2.actor.cpp#L946 FWIW

Reply:
> We might also consider fault injection here, but that's probably beyond the scope of this PR

Actually ::renameFile already does fault injection, so I think it's ok.

Reply:
yes, we're talking on slack about the fact that renameFile eats errors. I won't fix this in this PR though (the async version will throw an error)